packages/@azure/keyvault/lib/secretsClient.ts

```diff
@@ -318,12 +324,14 @@ export class SecretsClient {
 
   public async *getSecretVersions(
     secretName: string,
-    options?: RequestOptionsBase
+    options?: SecretClientGetPagedOptionalParams
   ): AsyncIterableIterator<Secret> {
     let currentSetResponse = await this.client.getSecretVersions(
       this.vaultBaseUrl,
       secretName,
-      options
+      { maxresults: options ? options.maxPageSize : undefined,

```

Comment:
What is the benefit of providing `maxPageSize` as an option to the user?
Does fetching smaller page size result in faster fetching of the data? Especially given that 25 is the max?

Also, now that we have the iterator, the page size should be a hidden implementation detail